### PR TITLE
feat: enable gosec and default linter set

### LINKED ISSUE #165: Add "make lint" target and add to "make test" target
Should enable golangci-lint with default linters + gosec.

See https://github.com/edgexfoundry/edgex-go/issues/3565

### PR DESCRIPTION
**Pull request by jpwhitemn**
addressed lint errors found in the code

Fixes #165

Signed-off-by: jpwhitemn <jpwhite_mn@yahoo.com>

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/device-rest-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [x] I am not introducing a new dependency (add notes below if you are)
- [x] I have added unit tests for the new feature or bug fix (if not, why?)
- [x] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [ ] I have opened a PR for the related docs change (if not, why?)  No docs to change for build feature
  <link to docs PR>

## Testing Instructions

run make lint target in the Makefile

### CODE REVIEW

**Review by bnevis-i — APPROVED** [review]

**Comment by bnevis-i** [comment]
recheck

**Comment by bnevis-i** [comment]
@cloudxxx8 Review requested.

File: driver/restdriver.go [hunk 1]
```diff
@@ -26,8 +26,8 @@ import (
 )
 
 type RestDriver struct {
-	logger      logger.LoggingClient
-	asyncValues chan<- *dsModels.AsyncValues
+	logger logger.LoggingClient
+	//asyncValues chan<- *dsModels.AsyncValues
```
Comment (lenny-intel) [inline driver/restdriver.go:30]: Remove commented out code.

File: driver/resthandler.go [hunk 2]
```diff
@@ -292,24 +295,23 @@ func (handler RestHandler) newCommandValue(resource model.DeviceResource, readin
 func checkUintValueRange(valueType string, val interface{}) error {
 	switch valueType {
 	case common.ValueTypeUint8:
-		valUint8 := val.(uint8)
-		if valUint8 <= math.MaxUint8 {
+		_, ok := val.(uint8)
+		if ok {
```
Comment (lenny-intel) [inline driver/resthandler.go:299]: Should be:
```suggestion
		if !ok {
```
And for the other also.

**Review by lenny-intel — CHANGES_REQUESTED** [review]

**Review by bnevis-i — APPROVED** [review]

**Review by lenny-intel — APPROVED** [review]

**Pull request merged**